internal/nodes/dequeue_ingestion.go

```diff
@@ -528,6 +530,17 @@ func handleL4SharedVipService(namespacedVipKey, key string, fullsync bool) {
 			if infraSettingAnnotation, ok := svcObj.GetAnnotations()[lib.InfraSettingNameAnnotation]; ok && infraSettingAnnotation != "" {
 				sharedVipInfraSetting = infraSettingAnnotation
 			}
+			if appProfileAnnotation, ok := svcObj.GetAnnotations()[lib.LBSvcAppProfileAnnotation]; ok && appProfileAnnotation != "" {
+				err := lib.CheckL4RefOnController(key, lib.AppProfile, appProfileAnnotation, avicache.SharedAVIClients())
+				if err != nil {
+					utils.AviLog.Warnf("key: %v, msg: Not processing L4 service", key)
+					isShareVipKeyDelete = true
+					break
+				} else {
+					appProfile = appProfileAnnotation
```

Comment:
Please remove this else statement since the following line is also doing the same assignment.